Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishHolidayParserConfiguration.cs

```diff
@@ -19,41 +19,110 @@ public EnglishHolidayParserConfiguration() : base()
+                { "usindependenceday", new string[]{ "usindependenceday" } },
+                { "independenceday", new string[]{ "independenceday" } },
+                { "bastilleday", new string[]{ "bastilleday" } },
+                { "halloweenday", new string[]{ "halloweenday" } },
+                { "allhallowday", new string[]{ "allhallowday" } },
+                { "allsoulsday", new string[]{ "allsoulsday" } },
+                { "guyfawkesday", new string[]{ "guyfawkesday" } },
+                { "veteransday", new string[]{ "veteransday" } },
+                { "christmaseve", new string[]{ "christmaseve" } },
+                { "newyeareve", new string[]{ "newyearseve" } },
             };
         }
 
         protected override IDictionary<string, Func<int, DateObject>> InitHolidayFuncs()
         {
             return new Dictionary<string, Func<int, DateObject>>(base.InitHolidayFuncs())
             {
-                {"yuandan", Yuandan},
-                {"chinesenational", ChineseNationalDay},
+                {"maosbirthday", MaoBirthday},
+                {"yuandan", NewYear},
+                {"teachersday", TeacherDay},
+                {"singleday", SinglesDay},
+                {"allsaintsday", HalloweenDay},
+                {"youthday", YouthDay},
+                {"childrenday", ChildrenDay},
+                {"femaleday", FemaleDay},
+                {"treeplantingday", TreePlantDay},
+                {"arborday", TreePlantDay},
+                {"girlsday", GirlsDay},
+                {"whiteloverday", WhiteLoverDay},
+                {"loverday", ValentinesDay},
+                {"christmas", ChristmasDay},
+                {"xmas", ChristmasDay},
+                {"newyear", NewYear},
+                {"newyearday", NewYear},
+                {"newyearsday", NewYear},
+                {"inaugurationday", InaugurationDay},
+                {"groundhougday", GroundhogDay},
+                {"valentinesday", ValentinesDay},
+                {"stpatrickday", StPatrickDay},
+                {"aprilfools", FoolDay},
+                {"stgeorgeday", StGeorgeDay},
+                {"mayday", Mayday},
+                {"cincodemayoday", CincoDeMayoday},
+                {"baptisteday", BaptisteDay},
+                {"usindependenceday", USAIndependenceDay},
+                {"independenceday", USAIndependenceDay},
+                {"bastilleday", BastilleDay},
+                {"halloweenday", HalloweenDay},
+                {"allhallowday", AllHallowDay},
+                {"allsoulsday", AllSoulsday},
+                {"guyfawkesday", GuyFawkesDay},
+                {"veteransday", Veteransday},
+                {"christmaseve", ChristmasEve},
+                {"newyeareve", NewYearEve},
             };
         }
         
-        private static DateObject Yuandan(int year) => new DateObject(year, 1, 1);
-        private static DateObject ChineseNationalDay(int year) => new DateObject(year, 10, 1);
```

Comment:
rename yuandan to newyear. Didn't extract Chinese National Day in English holidays. It is supported in Chinese holidays.